WordPress/openverse
# Upload Ingestion Server's TSV files to AWS S3 (skip tags)

## Fixes



Fixes #3912 by @krysal

## Description


Adds the function to upload the produced files of cleaned data to AWS S3 and delete them locally if the upload is successful.

## Testing Instructions

1. Set the required environment variables in your `.env` file
```
AWS_REGION="us-east-1"
AWS_ACCESS_KEY_ID="test_key"
AWS_SECRET_ACCESS_KEY="test_secret"
AWS_S3_ENDPOINT="http://s3:5000"
```
2. Spin up the services
```
just a && just c
```
3. Make sure you have the `openverse-catalog` bucket created in MinIO (or the one you'll be using; it can be configured with `OPENVERSE_BUCKET`):  http://localhost:5011/browser/
4. Make some rows of the image table in the catalog "dirty" by removing the protocol in one or more of the URLs fields (url, creator_url, or foreign_landing_url)
5. Run an image data refresh
```
# Optionally, delete the image index before to avoid the error of "index already exists"
just docker/es/delete-index image-init

just ingestion_server/ingest-upstream "image" "init"
```
6. Check the files are in the expected path in MinIO: http://localhost:5011/browser/openverse-catalog, and the URL values are **_not_** surrounded by quotation marks (they are unnecessary). 

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
